### ISSUE TITLE
REST-API locale is always default scope

### ISSUE THREAD

**Issue by jwundrak** [magento/magento2#19908]
### Preconditions (*)

1. Error occourds in 2.1.14, but I can reproduce this also in 2.3-develop
2. at least two store views. Default store-view locale != second store view
3. It doesn't matter, if you have one or multiple stores

### Steps to reproduce (*)
1. Configure stores. I use Store-View "default" with locale en_US and StoreView "second" with locale de_DE
2. Use something, that returns magento phrase (I wrote a test module here: https://github.com/jwundrak/magento2-api-locale )
3. Call rest API (with the module /rest/second/V1/localeTest)

### Actual result (*)

Output of the test module:
~~~
{
    "locale": "en_US",
    "test_translation": "Category",
    "store_code": "second",
    "store_locale": "de_DE"
}
~~~

### Expected result (*)

Output of the test module:
~~~
{
    "locale": "de_DE",
    "test_translation": "Kategorie",
    "store_code": "second",
    "store_locale": "de_DE"
}
~~~

### Possible solution

In `\Magento\Webapi\Controller\PathProcessor::process` I start an emulation of the locale: `$this->localeResolver->emulate($this->storeManager->getStore()->getId());` . I'm not sure, if this is the correct way, but works here in 2.1 and 2.3: https://gist.github.com/jwundrak/11ea7ca0b5c6cea57d3b7c8133548634

**Comment by magento-engcom-team** [comment]
Hi @jwundrak. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@jwundrak do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by jwundrak** [comment]
do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?

* [x]  yes
* [ ]  no

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by shikhamis11** [comment]
@jwundrak I tried the  same at magento 2.2 and 2.3 got the 2nd store locale

**Comment by magento-engcom-team** [comment]
@shikhamis11 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97267`, `MAGETWO-97268` were created

**Comment by jwundrak** [comment]
@shikhamis11 the expected and actual result was mixed up. I edit the description

**Comment by shikhamis11** [comment]
@jwundrak yes you are right I tried with ar_SA and en_US (default) got the result 
{
  "locale": "en_US",
  "test_translation": "Category",
  "store_code": "admin",
  "store_locale": "ar_SA"
}

**Comment by jwundrak** [comment]
@shikhamis11 what's with the patch in the gist. This resolves the error, but I don't know, if this way is good.
If so, I can prepare a pull request and write some tests

**Comment by shikhamis11** [comment]
@jwundrak I am not completely sure about the patch but if it fix the issue go for pull request

**Comment by alexander-aleman** [comment]
I created a plugin using your fix as a workaround [https://packagist.org/packages/xsarus/magento2-rest-api-locale](https://packagist.org/packages/xsarus/magento2-rest-api-locale)

**Comment by magento-engcom-team** [comment]
Hi @jwundrak. Thank you for your report. 
The issue has been fixed in magento/magento2#19913 by @jwundrak in 2.3-develop branch
Related commit(s):
 - [5abd3308539fd0449d9ffe22a2d826e1909cc074](https://github.com/magento/magento2/commit/5abd3308539fd0449d9ffe22a2d826e1909cc074)
 - [ef724a622b5a45cb8dbeb9ca5f5d5039af4affcf](https://github.com/magento/magento2/commit/ef724a622b5a45cb8dbeb9ca5f5d5039af4affcf)
 - [29b4b3920d413c2571ef2d8fd2fb1dba55f63d43](https://github.com/magento/magento2/commit/29b4b3920d413c2571ef2d8fd2fb1dba55f63d43)
 - [13ca4a23c2f34199caec25995d85ae4301268bd6](https://github.com/magento/magento2/commit/13ca4a23c2f34199caec25995d85ae4301268bd6)
 - [1cb3488686fb99814edcc0a898acc8a26f2ab16a](https://github.com/magento/magento2/commit/1cb3488686fb99814edcc0a898acc8a26f2ab16a)

The fix will be available with the upcoming 2.3.2 release.
